direct.go

```diff
@@ -213,19 +252,31 @@ func doCheck(client *http.Client, method string, expectedStatus int, u string) b
 // If it can't obtain a working masquerade within the given timeout, it will
 // return nil/false.
 func NewDirect(timeout time.Duration) (http.RoundTripper, bool) {
+	start := time.Now()
 	instance, ok := _instance.Get(timeout)
 	if !ok {
 		log.Errorf("No DirectHttpClient available within %v", timeout)
 		return nil, false
 	}
-	return instance.(http.RoundTripper), true
+	remaining := timeout - time.Since(start)
+
+	// Wait to be signalled that at least one masquerade has been vetted...
```

Comment:
Nice!

Reply:
Yeah, we'll definitely do a fair amount of manual testing with this release. I see that the current timeout in flashlight is 5 minutes. That seems pretty reasonable.